### DISCUSSION TITLE
Pick todays date in calendar

### DISCUSSION THREAD

**Discussion by Jerry3737** [mobile-dev-inc/Maestro#1759] [Q&A]
Hello, 

I wanted to ask for help. I am trying to automate a test case, where the user needs to pick a date from the calendar date picker. Given the circumstances of the test case and given that I want it to be relaunchable any day, I think the best approach might be to generate todays date with JS, then get the separate values for day, month and year and input them into the selector. The selector looks like this: 

- tapOn:
    id: "dateSelector2024-05-29"

Is this somehow possible? Would you be so kind and show me the right direction? Any help is appreciated. Thanks!

EDIT: So I can create a js file in my project, create the date there and use it as the file's output, which I can then use in maestro test yaml file, am I correct?

**Comment by kris-luminar** [comment]
Yes, that's fairly easy. Try something similar to the approach starting here
https://maestro.mobile.dev/examples/advanced-wikipedia-android#authentication
and here
https://maestro.mobile.dev/examples/advanced-wikipedia-android#javascript-scripts

but instead use date calculated in JavaScript
